sycl/plugins/level_zero/pi_level_zero.hpp

```diff
@@ -132,6 +132,17 @@ struct _pi_device : _ur_device_handle_t {
   using _ur_device_handle_t::_ur_device_handle_t;
 };
 
+// Structure describing the command queue descriptor when command lists are
+// created. This is used to ensure that the appropriate command list type is
+// reused from the context's cache.
+struct pi_command_list_and_desc_t {
+  ze_command_list_handle_t CmdList;
+  uint32_t Index{0};
+  ze_command_queue_flags_t Flags{0};
+  ze_command_queue_mode_t Mode{ZE_COMMAND_QUEUE_MODE_DEFAULT};
+  ze_command_queue_priority_t Priority{ZE_COMMAND_QUEUE_PRIORITY_NORMAL};
```

Comment:
why is it not simple a `ze_command_queue_desc_t`?

Reply:
and please add a comment that this is for IMM lists only

Reply:
An immediate command list is created using a queue descriptor, which contains ordinal, index, priority, mode, etc. Once the descriptor has been created we cannot recover any of the fields used to create it, i.e. ordinal, index, priority, mode etc.
To cache and reuse immediate command lists they must be compatible, i.e. the queue descriptor must have identical values. But because we can't recover the field values, we must sav e them all, and when deciding if an immediate commandlist can be taken from the cache, check if the queue descriptor is compatible.

Reply:
I meant that you would save the entire `ze_command_queue_desc_t` structure and not field-by-field

Reply:
This code in pi_level_zero.hpp is simply assigning default initial values to the fields.

Reply:
```suggestion
ze_command_queue_desc_t Desc;
```

Reply:
The ze_command_queue_desc_t is an opaque struct whose "public" fields are as listed above. There is no guarantee that those are the only fields, and that they are in that order. If declared like you suggest then the setting of a value of type pi_command_list_info_t will need to come from an initialized ze_command_queue_desc_t variable, whose members are initialized individually (instead of being an unnamed struct initializer).

Reply:
Ok, but I agree that it will look cleaner. Making that change.